src/tests/users/signup.test.js

```diff
@@ -1,30 +1,268 @@
+    return request({
+      query: `
+        mutation {
+          signup(record: {
+              firstName: "${fakeUser.firstName}"
+              lastName: "${fakeUser.lastName}"
+              email: "${fakeUser.email}"
+              password: "${fakeUser.password}"  
+              username: "${fakeUser.username}"
+              mentor: "${users[0]._id}"
+              skills: [${skills}]
+              knowledge: [${knowledge}]
+          }) {
+            record {
+              email
+            }
+          }
+        }
+      `
+    })
+      .expect((res) => {
+        expect(res.body).toHaveProperty('errors')
+        expect(Array.isArray(res.body.errors)).toBe(true)
+      })
+      .expect(200)
+  })
+
+  it('should make user a mentor', () => {
+    return request({
+      query: `
+        mutation {
+          updateUserById (_id: "${users[0]._id}", record: {
+            mentorshipCertified: true
+          }) {
+            record {
+              email
+            }
+          }
+        }
+      `
+    })
+      .expect((res) => {
+        expect(res.body).toHaveProperty('data.updateUserById.record')
+        expect(res.body.data.updateUserById.record.email).toStrictEqual(
+          users[0].email
+        )
+      })
+      .expect(200)
+  })
+
+  it('should not verify email with incorrect verification code', () => {
+    return request({
+      query: `
+        mutation {
+          verifyEmail (verificationCode: "XYZ") {
+            record 
+            error {
+              message
+            }
+          }
+        }
+      `
+    })
+      .expect((res) => {
+        expect(res.body).toHaveProperty('data.verifyEmail.error.message')
+        expect(res.body.data.verifyEmail.error.message).toStrictEqual(
+          'no user associated with this verification code'
+        )
+      })
+      .expect(200)
+  })
+
+  it('should verify email', () => {
+    return request({
+      query: `
+        mutation {
+          verifyEmail (verificationCode: "${users[0].verificationCode}") {
+            record
+          }
+        }
+      `
+    })
+      .expect((res) => {
+        expect(res.body).toHaveProperty('data.verifyEmail.record')
+        expect(res.body.data.verifyEmail.record).toStrictEqual(true)
+      })
+      .expect(200)
+  })
+
+  it('check email verified already', () => {
+    return request({
+      query: `
+        mutation {
+          verifyEmail (verificationCode: "${users[0].verificationCode}") {
+            record 
+            error {
+              message
+            }
+          }
+        }
+      `
+    })
+      .expect((res) => {
+        expect(res.body).toHaveProperty('data.verifyEmail.error.message')
+        expect(res.body.data.verifyEmail.error.message).toStrictEqual(
+          'user verified already'
+        )
+      })
+      .expect(200)
+  })
+
+  it('there are more than 0 mentors in database', () => {
```

Comment:
Im not sure what this is trying to test

Reply:
once mentor is made there should be more than 0 mentors in database.